src/auto_scheduler/transform_step.cc

```diff
@@ -51,129 +53,539 @@ const char* IteratorAnnotationString[] = {
     "tensorize"     // kTensorized = 11
 };
 
-/********** Reorder **********/
-ReorderStep::ReorderStep(int stage_id, const Array<Integer>& after_ids) {
-  auto node = make_object<ReorderStepNode>();
-  node->stage_id = stage_id;
-  for (const auto& x : after_ids) {
-    CHECK(x->IsInstance<IntImmNode>());
+Step StepReadFromRecord(dmlc::JSONReader* reader) {
```

Comment:
It would be better to have a step registration mechanism to establish a step lookup table from record prefix to the step node class, so that we can totally hide step node dispatching from developers. The potential issues in the current implementation is that 1) people may forget to update this function when adding a new step, and 2) there's no mechanism to check duplicated record prefix of two steps.

In addition, I feel this issue is important but I'm not sure if it's proper to address it in this PR. We may need versioning for records.

Reply:
1. Yes, we can add a registration mechanism. But I think it can be done in future PRs.
2. The current records do have a version record 
https://github.com/apache/incubator-tvm/blob/7ac78d96611ab3a069fae9eb5210e433eb240cc5/src/auto_scheduler/measure_record.cc#L314